packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp

```diff
@@ -358,27 +396,69 @@ namespace MueLu {
     LO num_bad_aggs = 0;
     MT worst_agg = 0.0;
 
+    MT mean_bad_agg = 0.0;
+    MT mean_good_agg  = 0.0;
+
+
     for (size_t i=0;i<agg_qualities->getLocalLength();++i) {
 
-      if (data[i] > good_agg_thresh) num_bad_aggs++;
+      if (data[i] > good_agg_thresh) {
+        num_bad_aggs++;
+        mean_bad_agg += data[i];
+      }
+      else {
+        mean_good_agg += data[i];
+      }
       worst_agg = std::max(worst_agg, data[i]);
-
     }
 
+
+    if (num_bad_aggs > 0) mean_bad_agg /= num_bad_aggs;
+    mean_good_agg /= agg_qualities->getLocalLength() - num_bad_aggs;
+
     if (num_bad_aggs == 0) {
-      GetOStream(Statistics1) << "All aggregates passed the quality measure. Worst aggregate had quality " << worst_agg << std::endl;
+      GetOStream(Statistics1) << "All aggregates passed the quality measure. Worst aggregate had quality " << worst_agg << ". Mean aggregate quality " << mean_good_agg << "." << std::endl;
```

Comment:
Same comment as before.  I don't think that a process `i !=root` will print, even if `num_bad_aggs>0` on proc `i`.

Reply:
Yup.

I don't think we have a way to control for output level w/o controlling processor count.

Reply:
Global reduce over `num_bad_aggs`?

Reply:
@cgcgcg That was what I was thinking.  O/W there's failure w/o a message.  (There's a joke in there somewhere.)